### ISSUE TITLE
Expose required descriptor data for at-rules

### ISSUE THREAD

**Issue by nzakas** [csstree/csstree#308]
Some at-rules, such as `@property` and `@font-face`, have required descriptors. `mdn-data` [exposes this info](https://github.com/mdn/data/blob/8ec6d5057d8b774c185e33fe684b5af4dc8f3333/css/at-rules.json#L22) but there's no way to get at it from the lexer.

Could we add a `required` field to the descriptor definition stored in `Lexer`?

**Comment by lahmatiy** [comment]
Yes, we can extract it in [lib/data.js](https://github.com/csstree/csstree/blob/master/lib/data.js) and incorporate it into the `Lexer` and syntax mix. The only challenge is that the descriptor's structure would become more complex. Currently, it's just a syntax string, but adding a `required` field would turn the descriptor definition into an object, requiring more extensive changes across the codebase. 

As a workaround, the `at-rule` definition could be extended with a `requiredDescriptors` field instead. However, this approach doesn’t seem ideal.

**Comment by nzakas** [comment]
I can take a stab at implementing this if you'd like.
